src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Carnivore || animal.hidden === true) {
+      return;
+    }
+
+    animal.health -= 50;
+
+    if (animal.health <= 0 && Animal.alive.includes(animal)) {
+      const index = Animal.alive.indexOf(
+        creature => creature === animal);
+
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
add this method to `Animal` class and use just filter method to remove animal from list

Reply:
Yeap, with filter is much easier, but why I should add bite method to Animal?